### ISSUE TITLE
After retry no messages delivered to consumer

### ISSUE THREAD

**Issue by tib13** [reactor/reactor-kafka#19]
Hi,

with SampleConsumer.java from your repo I can consume messages. 
After each restart it consumes the expected number or messages with right offset.

Problem occurs if a new RuntimeException occurs in map function. 
After retry (re-subscription) messages are not any more delivered to receiver. 
(After restart it works again until next exception&retry..., kafka 0.11)

Scenario like in SampleScenarios.java:
```
return KafkaReceiver.create(receiverOptions(Collections.singletonList(topic))
  .commitInterval(Duration.ZERO))
  .receive()
  .map(r -> { 
      if (atomicCounter.getAndIncrement() > 10)  {
          throw new RuntimeException("TEST");
      }
      r.receiverOffset().commit().block();
      return r;
  }
  .retry()
  .doOnCancel(() -> close())
  .subscribe(r ->System.out.print(r.value()));

**Comment by bsideup** [comment]
@rajinisivaram @smaldini we observe the same issue, do you have any update on it? 
Seems to not comply with the Reactive spec

**Comment by bsideup** [comment]
After a discussion with @smaldini it seems that there is a workaround:
```
KafkaReceiver receiver =  KafkaReceiver.create(receiverOptions(Collections.singletonList(topic))
  .commitInterval(Duration.ZERO));
return Flux.defer(receiver::receive)
  .map(r -> { 
      if (atomicCounter.getAndIncrement() > 10)  {
          throw new RuntimeException("TEST");
      }
      r.receiverOffset().commit().block();
      return r;
  }
  .retry()
  .subscribe(r ->System.out.print(r.value()));
```

**Comment by mradha1** [comment]
Thank for the work around...Any updates on fixing the retry?

**Comment by FcoJavierSainz** [comment]
After one retry its stop again :(

**Comment by OlegDokuka** [comment]
@FcoJavierSainz Can you please doublecheck that issue with the snapshot version (1.0.1.BUILD-SNAPSHOT)

**Comment by FcoJavierSainz** [comment]
@OlegDokuka It is working for me :) Thanks

**Comment by OlegDokuka** [comment]
@FcoJavierSainz, amazing! Thank you for doublechecking that!!! Close since fixed by #35

**Comment by jimhorng** [comment]
Seemed I'm facing the similar problem on using `1.0.1.RELEASE`

- The 1st attempt it got below, which seems expected since I intentionally throw a `RuntimeException`
```
doFinally: signalType:cancel
doOnError:class java.lang.RuntimeException
doOnError:class reactor.core.Exceptions$ReactorRejectedExecutionException
```
- it does retry (show client config initiating in log), but in 2nd attempt and after, it got below error and does NOT receive message anymore :(
```
doFinally: signalType:onError
doOnError:class reactor.core.Exceptions$ReactorRejectedExecutionException
```
my code looks something like this:
```java
KafkaReceiver<String, byte[]> kafkaReceiver = KafkaReceiver.create(options);
Flux<ReceiverRecord<String, byte[]>> kafkaFlux = kafkaReceiver.receive();
return kafkaFlux
        .doFinally(signalType -> {
            log.info("doFinally: signalType:{}", signalType.toString());
        })
        .map(record -> {
            ReceiverOffset offset = record.receiverOffset();
            processEvent(record, offset);
            offset.acknowledge();
            return record;
        })
        .doOnError(error -> {
            log.info("doOnError:{}", error.getClass());
            this.counter.set(0);
        })
        .retry(3)
        .subscribe()
```

Please advise, thanks :)

**Comment by ligasgr** [comment]
I'm also seeing the same issue on 1.0.1. 
The defer() workaround does work but that is only a workaround.

**Comment by jli207** [comment]
We are also seeing this issue. Will this get fixed any time soon? Thanks.

**Comment by hfgbarrigas** [comment]
Any news on this bug? A bit critical since it stalls consumers and forces a restart.

**Comment by abhimanyuseth-kiva** [comment]
I had same issue and found this solution on StackOverflow worked for me
https://stackoverflow.com/questions/54984724/reactor-kafka-at-least-once-handling-failures-and-offsets-in-multi-partition

**Comment by abhimanyuseth** [comment]
I had same issue and found this solution on StackOverflow worked for me
https://stackoverflow.com/questions/54984724/reactor-kafka-at-least-once-handling-failures-and-offsets-in-multi-partition

**Comment by billydh** [comment]
Will this be fixed in `1.3 RELEASE`?

**Comment by bsideup** [comment]
Should be fixed in `1.3.0.RELEASE`
